### ISSUE TITLE
Deleting a server in the LSP preferences pane when only one is present raises exceptions

### ISSUE THREAD

**Issue by CAM-Gerlach** [spyder-ide/spyder#8816]
## Problem Description

With default preferences, deleting the only server listed in the LSP preference pane will cause an exception immediately, and further exceptions when clicking anywhere within the pane. Clearly, these routines are expecting at least one server to be present, and not handling the case if none are.

### What steps reproduce the problem?

1. Select only (default) LSP server in the list in the LSP preference pane.
2. Press ``Delete currently selected server``

### Paste Traceback/Error Below (if applicable)

Immediately on clicking delete

```python-traceback

  File "C:\Users\C. A. M. Gerlach\Documents\dev\SpyderDev\spyder\spyder\preferences\languageserver.py", line 646, in delete_server
    self.table.delete_server(idx)
  File "C:\Users\C. A. M. Gerlach\Documents\dev\SpyderDev\spyder\spyder\preferences\languageserver.py", line 554, in delete_server
    self.adjust_cells()
  File "C:\Users\C. A. M. Gerlach\Documents\dev\SpyderDev\spyder\spyder\preferences\languageserver.py", line 524, in adjust_cells
    self.setColumnWidth(CMD, max(names))
ValueError: max() arg is an empty sequence

```

Clicking in the pane thereafter:

```python-traceback
  File "C:\Users\C. A. M. Gerlach\Documents\dev\SpyderDev\spyder\spyder\preferences\languageserver.py", line 605, in mouseDoubleClickEvent
    self.show_editor()
  File "C:\Users\C. A. M. Gerlach\Documents\dev\SpyderDev\spyder\spyder\preferences\languageserver.py", line 561, in show_editor
    server = self.source_model.row(idx)
  File "C:\Users\C. A. M. Gerlach\Documents\dev\SpyderDev\spyder\spyder\preferences\languageserver.py", line 478, in row
    return self.servers[row_num]
IndexError: list index out of range
```

## Versions

* Spyder version: 4.0.0.dev0 a411d61d7
* Python version: 3.7.2
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 8.1

### Dependencies

```
pygments >=2.0    :  2.3.1 (OK)
qdarkstyle >=2.6.4:  2.6.5 (OK)
sphinx >=0.6.6    :  1.8.4 (OK)
pyls >=0.19.0     :  0.21.5 (OK)
nbconvert >=4.0   :  5.4.0 (OK)
pandas >=0.13.1   :  0.24.1 (OK)
numpy >=1.7       :  1.15.4 (OK)
sympy >=0.7.3     :  1.3 (OK)
cython >=0.21     :  0.29.4 (OK)
qtconsole >=4.2.0 :  4.4.3 (OK)
IPython >=4.0     :  7.2.0 (OK)
matplotlib >=2.0.0:  3.0.2 (OK)
pylint >=0.25     :  2.2.2 (OK)
```

**Comment by ccordoba12** [comment]
This is too obvious to be stated (sorry, but it is). PR #8647 will prevent this by making only allowing users to add new users and having PyLS as the default one.

**Comment by CAM-Gerlach** [comment]
> This is too obvious to be stated (sorry, but it is).

Sorry, are you saying that the error is obvious (if it is, I'm not sure why its apparently been present unfixed and undocumented for over half a year :) )?  Or do you mean that that its obvious that #8647 would fix it? When I tested that PR for this issue, the error was still present exactly the same as I report here, and it wasn't documented anywhere on that PR (or anywhere else I could find on this repo), so I'm not sure I understand how it should have been so obvious, sorry.

Furthermore, several times now you've reprimanded me for not opening separate issues for issues not introduced by a PR, so I didn't want to make that mistake again. Other than a few extra seconds to skim this issue and add a ``Fixes`` line to the PR, I'm not really sure I see what downside of reporting it would outweigh the risk of an easily reproducible bug going undocumented and unfixed.

**Comment by ccordoba12** [comment]
> Sorry, are you saying that the error is obvious

Yes, it is obvious because PyLS is the only server shown in that config page and we're providing completions for python files by depending on the PyLS package. So if you remove that entry everything breaks down.

> if it is, I'm not sure why its apparently been present unfixed and undocumented for over half a year :)

Because we haven't had time to reach to the point where we can fix this, but I mentioned on issue #7724 that we needed to rework that config page, and I opened #7724 right after merging the LSP architecture.

----

Look, I know you're are well intended when opening these issues, but for next time please consult with me on Gitter if there's a need for them or not.

**Comment by CAM-Gerlach** [comment]
> Yes, it is obvious because PyLS is the only server shown in that config page and we're providing completions for python files by depending on the PyLS package. So if you remove that entry everything breaks down.

Then it shouldn't be possible to remove it, exactly as you've said. However, there was no indication in either #7724 or #8647 that this was going to be added to the latter, and I wanted to make sure it was addressed. Next time I will ask on Gitter as you've requested.

> for next time please consult with me on Gitter if there's a need for them or not.

Will do for all future issues to avoid this happening again :+1:
